lambda/simplif.ml

```diff
@@ -226,9 +230,12 @@ let simplify_exits lam =
   | Lmutlet(kind, v, l1, l2) ->
       Lmutlet(kind, v, simplif ~try_depth l1, simplif ~try_depth l2)
   | Lletrec(bindings, body) ->
-      Lletrec(
-        List.map (fun (v, clas, l) -> (v, clas, simplif ~try_depth l)) bindings,
-        simplif ~try_depth body)
+      let bindings =
+        List.map (fun { id; rkind; def } ->
+            { id; rkind; def = simplif ~try_depth def })
```

Comment:
... maybe there should be a `map_rec_binding : (lambda -> lambda) -> rec_binding -> rec_binding` function in Lambda?

Reply:
Or maybe we could rewrite all of these `Simplif` functions in terms of the existing `Lambda` iterators.